### ISSUE TITLE
Flaky test - The node backend is unreachable at the moment.

### ISSUE THREAD

**Issue by KtorZ** [cardano-foundation/cardano-wallet#2320]
# Context
 
**Bug report:** https://jira.iohk.io/browse/ADP-647

 

Saw only once, could just be us overloading the node due to parallel tests, especially in this region

- https://hydra.iohk.io/build/4800986
- https://hydra.iohk.io/build/5279297

# Test Case

- STAKE_POOLS_QUIT_02 from `src/Test/Integration/Scenario/API/Shelley/StakePools.hs:433`
- STAKE_POOLS_QUIT_01x

# Failure / Counter-example

```
  1) API Specifications, SHELLEY_STAKE_POOLS, STAKE_POOLS_QUIT_02 - Passphrase must be correct to quit
       uncaught exception: RequestException
       DecodeFailure "{\"code\":\"network_unreachable\",\"message\":\"The node backend is unreachable at the moment. Trying again in a bit might work.\"}"
```
```
  src/Test/Integration/Scenario/API/Shelley/StakePools.hs:850:9: 
  2) API Specifications, SHELLEY_STAKE_POOLS, STAKE_POOLS_QUIT_01x - Fee boundary values, STAKE_POOLS_QUIT_01xx - I can quit if I have enough to cover fee
       uncaught exception: RequestException
       DecodeFailure "{\"code\":\"network_unreachable\",\"message\":\"The node backend is unreachable at the moment. Trying again in a bit might work.\"}"
```

It seems that most recent failures are always preceded by:

```
  src/Test/Integration/Framework/DSL.hs:1471:16: 
  1) API Specifications, SHELLEY_WALLETS, WALLETS_LIST_01 - Wallets are listed from oldest to newest
       uncaught exception: ErrorCall
       getFromResponse failed to get item
       CallStack (from HasCallStack):
         error, called at src/Test/Integration/Framework/DSL.hs:1471:16 in cardano-wallet-core-integration-2020.12.21-AAYZFneZURcKNEp0gMVmAX:Test.Integration.Framework.DSL

  To rerun use: --match "/API Specifications/SHELLEY_WALLETS/WALLETS_LIST_01 - Wallets are listed from oldest to newest/"
```

So, it might be the one we ought to investigate. 

# Resolution

See https://jira.iohk.io/browse/ADP-647

**Comment by rvl** [comment]
@KtorZ That error message is a big red herring.
There is no evidence that the node backend is unreachable.
All it means is that some LSQ returned some kind of failure.
We should review and merge #2419 as a matter of urgency, so that we stop seeing this error message, and instead have enough information logged so that we can debug the failure.

**Comment by KtorZ** [comment]
@rvl I agree. The error message isn't really helpful as it is but I am pretty convinced that it would suffice to retry the `stakeDistribution` query after re-acquiring a more recent point.

**Comment by rvl** [comment]
I opened an issue - ADP-647.

**Comment by Anviking** [comment]
Seems odd that this failure only appeared recently though

**Comment by Anviking** [comment]
Let's keep this open still.

**Comment by rvl** [comment]
This should be closed because the tests fail due to an actual bug not a flaky test.
Also the error message has been improved, so the title of this issue is no longer relevant.

**Comment by Anviking** [comment]
I think it's useful that new failures end up in the same group as the previous 16 failures, such that people see that this is a common problem, and don't create new tickets for it.

It's still a test failure, even if it is also a bug.

**Comment by piotr-iohk** [comment]
Would this be a relevant bug report? -> https://jira.iohk.io/browse/ADP-647

**Comment by Anviking** [comment]
Yes

**Comment by rvl** [comment]
OK

**Comment by Anviking** [comment]
Should be fixed now.
